src/Vault.sol

```diff
@@ -1207,45 +1199,48 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
-   * @return uint256 Exchange rate
+   * @notice Calculates the exchange rate between the amount of underlying assets withdrawable from the YieldVault
+   *         and the total supply of shares minted by this Vault.
+   * @dev The initial exchange rate is 1, representing 1 unit of underlying asset.
+   * @dev When the Vault is collateralized, Vault shares are minted at a 1:1 ratio based on the user's deposited underlying assets.
+   *      The total supply of shares corresponds directly to the total amount of underlying assets deposited into the YieldVault.
+   *      Users have the ability to withdraw only the quantity of underlying assets they initially deposited,
+   *      without access to any of the accumulated yield within the YieldVault.
+   * @dev When the Vault is undercollateralized, the exchange rate diminishes below 1.
+   *      Withdrawals can be made by users for their corresponding deposit shares,
+   *      and any remaining unclaimed yield is distributed proportionally among depositors.
+   * @return uint256 Current exchange rate
    */
-  function _currentExchangeRate() internal view returns (uint256) {
-    uint256 _totalSupplyAmount = _totalSupply();
-    uint256 _depositedAssets = _convertToAssets(
-      _totalSupplyAmount,
-      _lastRecordedExchangeRate,
-      Math.Rounding.Down
-    );
-
+  function _exchangeRate() internal view returns (uint256) {
+    uint256 _depositedAssets = _totalSupply();
     uint256 _withdrawableAssets = _yieldVault.maxWithdraw(address(this));
 
-    // If the Vault is collateralized, yield is excluded from the withdrawable amount
-    // and users can only withdraw the amount of underlying assets they deposited.
-    // Otherwise, any unclaimed yield is included and shared proportionally amongst depositors.
-    if (_withdrawableAssets > _depositedAssets) {
-      _withdrawableAssets = _depositedAssets;
+    // If the Vault is collateralized, users can only withdraw the amount of underlying assets they deposited.
+    // An exchange rate of 1 is returned to exclude the amount of yield generated by the YieldVault.
+    if (_withdrawableAssets >= _depositedAssets) {
+      return _assetUnit;
     }
 
-    if (_totalSupplyAmount != 0 && _withdrawableAssets != 0) {
-      return _withdrawableAssets.mulDiv(_assetUnit, _totalSupplyAmount, Math.Rounding.Down);
+    // Otherwise, users can withdraw their corresponding deposit shares and
+    // any unclaimed yield is factored in and distributed proportionally among depositors.
+    if (_depositedAssets != 0 && _withdrawableAssets != 0) {
+      return _withdrawableAssets.mulDiv(_assetUnit, _depositedAssets, Math.Rounding.Down);
     }
 
-    return _assetUnit;
+    // Case when `_withdrawableAssets == 0` but `_depositedAssets != 0`.
+    // The Vault is entirely undercollateralized
+    // or the YieldVault has paused withdrawals and shares can't be redeemed.
+    return 0;
```

Comment:
Now that there is no historic `_lastRecordedExchangeRate`, it doesn't feel like this function serves a strong purpose, and it's doubling the number of `mulDiv` that `convertToShares` and `convertToAssets` have to do.

I imagine this function could be refactored into a separate function that checks available assets:

```
function _collateral() {
    uint256 _depositedAssets = _totalSupply();
    uint256 _withdrawableAssets = _yieldVault.maxWithdraw(address(this));

    // If the Vault is collateralized, users can only withdraw the amount of underlying assets they deposited.
    // An exchange rate of 1 is returned to exclude the amount of yield generated by the YieldVault.
    if (_withdrawableAssets >= _depositedAssets) {
      return _depositedAssets;
    }
    return _withdrawableAssets;
}
```

Then in your `convertToShares` and `convertToAssets` you're only doing 1 `mulDiv`:

```
function _convertToShares(
    uint256 _assets,
    Math.Rounding _rounding
  ) internal view virtual override returns (uint256) {
    uint totalSupplyShares = _totalSupply();
    return
      (_assets == 0 || totalSupplyShares == 0)
        ? _assets
        : _assets.mulDiv(totalSupplyShares, _collateral(), _rounding);
  }
```

and vice versa for assets.

This reduces the `mulDiv`, and makes the exchange rate clearer I think (less logic).

Reply:
But then once the Vault becomes undercollateralized, any yield that has not been claimed won't be withdrawable since we wouldn't include it in the exchange rate calculation.

Reply:
Yes we would; the _collateral() would be equal to _withdrawableAssets because it's undercollateralized.

Reply:
Added the `_collateral()` function in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/18/commits/81fe36a5c670947219d63287f4e205cc7d2593f6